snuba/api/query.py

```diff
@@ -132,7 +132,7 @@ def raw_query(
                         query_settings["max_threads"] = 1
 
                     try:
-                        result = NativeDriverReader(client).execute(
+                        result = reader.execute(
```

Comment:
`result` here is actually `Any` — _not_ `Result` as you'd expect from [`Reader.execute`](https://github.com/getsentry/snuba/blob/a317201ae7e8617b584021fee98a99e611540688/snuba/reader.py#L39-L45) due to the right hand side of the assignment statement having a return type of `Any`: https://github.com/getsentry/snuba/blob/bac8cf0189d7e947d002b29900176b990bd712a4/snuba/api/query.py#L91

I couldn't figure out why all the mutations on `result` below were not causing type errors, and that's why. They wouldn't be new ones, since this was always using a `Reader`. This function is promising a return type that it is unable to guarantee, and something we should probably address at some point.